core/src/builtin_widgets/theme/custom_styles.rs

```diff
@@ -24,20 +22,24 @@ macro_rules! fill_custom_style {
 }
 
 pub trait CustomStyle: Sized + 'static {
+  fn default_style(ctx: &BuildCtx) -> Self;
+
   #[inline]
-  fn of<'a>(ctx: &'a BuildCtx) -> Ref<'a, Self> {
+  fn of<'a>(ctx: &'a BuildCtx) -> &'a Self {
     let tid = TypeId::of::<Self>();
     let c = ctx.find_cfg(|t| match t {
       Theme::Full(t) => t.custom_styles.themes.get(&tid),
       Theme::Inherit(i) => i.custom_styles.as_ref().and_then(|c| c.themes.get(&tid)),
     });
-    c.and_then(|c| Ref::filter_map(c, |c| c.downcast_ref::<Self>()).ok())
-      .unwrap_or_else(|| {
-        panic!(
-          "The custom theme({}) is not init in theme, use it after init.",
-          std::any::type_name::<Self>()
-        )
-      })
+
+    c.unwrap_or_else(|| {
+      let style = Self::default_style(ctx);
+      let Theme::Full(app_theme) = ctx.app_theme_mut() else { unreachable!() };
```

Comment:
Why need `else` here?

Reply:
Only a `let else` syntax here to only accept a full theme, then I can omit an if else branch.